app/src/lib/stores/app-store.ts

```diff
@@ -5020,7 +5027,10 @@ export class AppStore extends TypedBaseStore<IAppState> {
   public async _createStash(
     repository: Repository,
     branchName: string,
-    removePreviousStash: boolean = true
+    preAction:
```

Comment:
Open to suggestions on better names for these two callbacks. My intent is to give the the calling code more control over what happens before and after a stash entry has been created.

Reply:
Rather than using `pre` and `post` here, I think these could be more explicit:

 - `onPreviousStashFound`
 - `onNewStashCreated`

This means they both can use the same signature, and it becomes a bit more intuitive than "before" and "after" in terms of what these are intended for.

Reply:
`onPreviousStashFound`'s signature actually can't be different since `getLastDesktopStashEntryForBranch` returns `null` if there is no previous stash. 

https://github.com/desktop/desktop/blob/139dad99ac256d6a9d74ef31614e95fa149be96d/app/src/lib/stores/app-store.ts#L5044-L5050

`onNewStashCreated`'s signature only gets a stash entry for sure because I verify it's there because it really should be there after just being created.

https://github.com/desktop/desktop/blob/139dad99ac256d6a9d74ef31614e95fa149be96d/app/src/lib/stores/app-store.ts#L5054-L5066

Reply:
> `onPreviousStashFound`'s signature actually can't be different since `getLastDesktopStashEntryForBranch` returns `null` if there is no previous stash.

Sure, but the only usage I found in the codebase of this parameter was performing some work _only if the previous stash was found_. That felt like an implicit part of the contract of the API: "register a callback to do some work only when a previous stash was found". 

What if we extracted that "function returning promise to its own type alias"?

```ts
export type StashCallback = (stashEntry: IStashEntry) => Promise<void>
```

We could make `AppStore._createStash` only run the callback if both the callback was set and it found a valid previous stash (hence the name `onPreviousStashEntryFound`)

```ts
    if (onPreviousStashEntryFound != null) {
      const previousStashEntry = await getLastDesktopStashEntryForBranch(
        repository,
        branchName
      )

      if (previousStashEntry !== null) {
        await onPreviousStashEntryFound(previousStashEntry)
      }
    }
```

And we could tidy up the signatures to be a bit more clearer:

```ts
  public async _createStash(
    repository: Repository,
    branchName: string,
    onPreviousStashEntryFound: StashCallback | null,
    onNewStashCreated: StashCallback | null
  ) {
```

Reply:
Addressed in 71059d3.